src/hooks/useBurgerMenu.ts

```diff
@@ -1,19 +1,27 @@
-import { useContext, useEffect } from 'react'
-import { BurgerMenuShowContext } from '../components/Context/BurgerMenuShowContext'
+import { useContext, useEffect} from 'react'
 import { BurgerMenuStatusContext } from '../components/Context/BurgerMenuStatusContext'
+import { useSearchParams } from './useSearchParams'
 
 export function useBurgerMenu() {
-  const burgerShow = useContext(BurgerMenuShowContext)
   const burgerMenu = useContext(BurgerMenuStatusContext)
+  const {type, status, searching, timeFrame} = useSearchParams()
 
   useEffect(() => {
-    burgerShow?.setShow(true)
-  
-    return () => {
-      burgerShow?.setShow(false)
-      burgerMenu?.setStatus(false)
-    };
-  }, []);
+      const filtering =  !!(type || status || timeFrame && timeFrame.length > 0)
+      burgerMenu?.setStatus({
```

Comment:
I'd use here `burgerMenu.setStatus(prevState => ({ ...prevState, ...rest })))`, also burgerMenu.status is not part of the dependencies of this effect, this can cause this effect to be outdated.

Reply:
👌 with `burgerMenu.setStatus(prevState => ({ ...prevState, ...rest })))`!

The status is nor part of the dependencies because this effect is meant to calculate the initial status for the burger menu based on the search params (if any!), but it should run once :)